src/commercetools/commercetools.service.ts

```diff
@@ -0,0 +1,409 @@
+import { Injectable, Logger } from '@nestjs/common';
+import {
+  Cart as CommerceToolsCart,
+  Order as CommerceToolsOrder,
+  Product,
+  TaxCategory,
+} from '@commercetools/platform-sdk';
+import { CommercetoolsConnectorService } from './commercetools-connector.service';
+import sleep from '../misc/sleep';
+import { Cart, ProductToAdd, Order } from '../integration/types';
+import { CustomTypesService } from './custom-types/custom-types.service';
+import {
+  OrdersItem,
+  RedemptionsRedeemStackableResponse,
+  StackableRedeemableResponse,
+  StackableRedeemableResultDiscountUnit,
+} from '@voucherify/sdk';
+import { getCommercetoolstCurrentPriceAmount } from './utils/getCommercetoolstCurrentPriceAmount';
+import { CUSTOM_FIELD_PREFIX } from '../consts/voucherify';
+import { CartAction } from './cartActions/CartAction';
+import { ConfigService } from '@nestjs/config';
+import { StoreData } from '../integration/integration.service';
+import { TaxCategoriesService } from './tax-categories/tax-categories.service';
+import { deleteObjectsFromObject } from './utils/deleteObjectsFromObject';
+import flatten from 'flat';
+import { getCouponsLimit } from '../voucherify/voucherify.service';
+import { ActionBuilder } from './cartActionsBuilder';
+import { getMaxCartUpdateResponseTimeWithoutCheckingIfApiExtensionTimedOut } from './utils/getMaxCartUpdateResponseTimeWithoutCheckingIfApiExtensionTimedOut';
+import { translateCtCartToCart } from './utils/mappers/translateCtCartToCart';
+import { getPriceSelectorFromCtCart } from './utils/mappers/getPriceSelectorFromCtCart';
+import { translateCtOrderToOrder } from './utils/mappers/translateCtOrderToOrder';
+import { CartDiscountApplyMode, CartResponse, PriceSelector } from './types';
+
+interface ProductWithCurrentPriceAmount extends Product {
+  currentPriceAmount: number;
+  unit: StackableRedeemableResultDiscountUnit;
+  item: OrdersItem;
+}
+
+type handlerCartUpdate = (
+  cart: Cart,
+  storeActions?: StoreData,
+  helperToGetProductsFromStore?: any,
```

Comment:
smart

Reply:
type is 'any' because its store dependent. 

for example we need type PriceSelector = {
  country: string;
  currencyCode: string;
  customerGroup: CustomerGroupReference;
  distributionChannels: ChannelReference[];
};
but another store may need something else or nothing at all.